labs/src/main/scala/org/scalalabs/basic/lab03/PatternMatchingExercise.scala

```diff
@@ -47,7 +53,15 @@ object PatternMatchingExercise {
    *    - anything else, the function result is "Some Scala class"
    */
   def matchOnInputType(in: Any): String = {
-    error("fix me")
+    in match {
+      case s:String => "A string with length " + s.length
+      case i:Int if i > 0 => "A positive integer"
+      case p:Person => "A person with name: " + p.name
+      case null => "A null value"
+      case seq:Seq[_] if seq.length > 10 => "Seq with more than 10 elements"
+      case seq:Seq[_] if seq.length > 3 => "first: " + seq.apply(0).toString() + ", second: " + seq.apply(1).toString() + ", rest: " + seq.slice(2, seq.length)
+      case default => "Some Scala class"
+    }
```

Comment:
In this case _ works fine. I'd use default if you want to actually use the default value for something. For instance, you could probably do default.getClass() but I doubt _.getClass() would compile (though who knows, maybe it would)